jwql/utils/engineering_database.py

```diff
@@ -0,0 +1,174 @@
+        query_single_mnemonic(mnemonic_identifier, start_time, end_time)
+
+    Required arguments:
+
+    ``mnemonic_identifier`` - String representation of a mnemonic name.
+    ``start_time`` - astropy.time.Time instance
+    ``end_time`` - astropy.time.Time instance
+
+Notes
+-----
+    This module is built on top of astroquery.mast and uses JWST-specific MAST services.
+    The user has to provide a valid MAST authentication token.
+
+"""
+import astropy
+from astropy.table import Table
+from astroquery.mast import Mast
+
+from .utils import get_config
+
+# should use oauth.register_oauth()?
+settings = get_config()
+mast_token = settings['mast_token']
+
+Mast.login(token=mast_token)
+
+# could eventually be moved to constants.py
+mast_edb_timeseries_service = 'Mast.JwstEdb.GetTimeseries.All'
+mast_edb_dictionary_service = 'Mast.JwstEdb.Dictionary'
+mast_edb_mnemonic_service = 'Mast.JwstEdb.Mnemonics'
+
+
+class EdbMnemonic():
+    """Class to hold and manipulate results of EngDB queries."""
+
+    def __init__(self, mnemonic_identifier, start_time, end_time, data, meta, info):
+        """Populate attributes."""
+        self.mnemonic_identifier = mnemonic_identifier
+        self.start_time = start_time
+        self.end_time = end_time
+        self.data = data
+        self.meta = meta
+        self.info = info
+
+    def __str__(self):
+        """Return string describing the instance."""
+        return 'EdbMnemonic {} with {} records between {} and {}'.format(
+            self.mnemonic_identifier, len(self.data), self.start_time.isot,
+            self.end_time.isot)
+
+    # def interpolate(self, date, **kwargs):
+    #     """Interpolate value at times specified in data argument."""
+    #     raise NotImplementedError
+
+
+def process_mast_service_request_result(result):
+    """Parse the result of a MAST EDB query into a astropy data table and a meta dictionary.
+
+    Parameters
+    ----------
+    result : list of requests.models.Response instances
+        The object returned by a call to Mast.service_request_async
+
+    Returns
+    -------
+        data : astropy.table.Table
+            Table representation of the returned data.
+        meta : dict
+            Additional information returned by the query
+
+    """
+    json_data = result[0].json()
+    if json_data['status'] != 'COMPLETE':
+        print(json_data['status'])
+        raise RuntimeError('mnemonic query did not complete.')
+
+    try:
+        # timestamp-value pairs in the form of an astropy table
+        data = Table(json_data['data'])
+    except KeyError:
+        raise RuntimeError('Query did not return any data.')
+
+    # collect meta data
+    meta = {}
+    for key in json_data.keys():
+        if key.lower() != 'data':
+            meta[key] = json_data[key]
+
+    return data, meta
+
+
+def get_all_mnemonic_identifiers():
+    """Return identifiers and meta data for all mnemonics in the DMS engineering database."""
```

Comment:
@Johannes-Sahlmann Could you add appropriate `Returns` section in this docstring?

Reply:
Done.